.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,41 @@
+name: Benchmark
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths-ignore:
+      - '.buildkite/**'
+      - '*.md'
+      - 'LICENSE'
+      - 'CODEOWNERS'
+      - 'renovate.json'
+      - 'introspection-engine/**'
```

Comment:
nitpick: we could ignore more in this case (migration-engine/**, lib/**), but of course we'd have to think about un-ignoring whenever we add benchmarks there.

Reply:
I leave that to the engineers to polish later - or just add benchmarks everywhere so we don't need to 😆